starknet-foundry/crates/cast/src/starknet_commands/account/create.rs

```diff
@@ -0,0 +1,187 @@
+use crate::helpers::constants::OZ_CLASS_HASH;
+use crate::helpers::scarb_utils::get_property;
+use anyhow::{anyhow, Context, Result};
+use camino::Utf8PathBuf;
+use cast::{get_network, parse_number, print_formatted};
+use clap::Args;
+use rand::rngs::OsRng;
+use rand::RngCore;
+use scarb::ops::find_manifest_path;
+use serde_json::json;
+use starknet::accounts::{AccountFactory, OpenZeppelinAccountFactory};
+use starknet::core::types::FieldElement;
+use starknet::core::utils::get_contract_address;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+use starknet::signers::{LocalWallet, SigningKey};
+use std::fs::OpenOptions;
+use std::io::Write;
+use toml::Value;
+
+#[derive(Args, Debug)]
+#[command(about = "Create an account with all important secrets")]
+pub struct Create {
+    /// Account name under which secrets are going to be saved
+    #[clap(short, long)]
+    pub name: String,
+
+    /// Salt for the address
+    #[clap(short, long)]
+    pub salt: Option<FieldElement>,
+
+    // If passed, a profile with corresponding data will be created in Scarb.toml
+    #[clap(short, long)]
+    pub add_profile: bool,
+    // TODO: think about supporting different account providers
```

Comment:
if this is not going to be added in this PR, let's create an issue for that (we can link it here)